src/dodal/devices/pressure_jump_cell.py

```diff
@@ -0,0 +1,236 @@
+from enum import Enum
+
+from ophyd_async.core import (
+    ConfigSignal,
+    DeviceVector,
+    SignalR,
+    StandardReadable,
+)
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_rw
+
+
+class PumpState(str, Enum):
+    MANUAL = "Manual"
+    AUTO_PRESSURE = "Auto Pressure"
+    AUTO_POSITION = "Auto Position"
+
+
+class BusyState(str, Enum):
+    IDLE = "Idle"
+    BUSY = "Busy"
+
+
+class TimerState(str, Enum):
+    TIMEOUT = "TIMEOUT"
+    COUNTDOWN = "COUNTDOWN"
+
+
+class StopState(str, Enum):
+    CONTINUE = "CONTINUE"
+    STOP = "STOP"
+
+
+class ValveControlRequest(str, Enum):
+    OPEN = "Open"
+    CLOSE = "Close"
+    RESET = "Reset"
+    ARM = "Arm"
+    DISARM = "Disarm"
+
+
+class PumpMotorControlRequest(str, Enum):
+    ENABLE = "Enable"
+    DISABLE = "Disable"
+    RESET = "Reset"
+    FORWARD = "Forward"
+    REVERSE = "Reverse"
+
+
+class PumpMotorDirectionState(str, Enum):
+    ZERO = "0"
+    FORWARD = "Forward"
+    REVERSE = "Reverse"
+    THREE = "3"
+    FOUR = "4"
+    FIVE = "5"
+    SIX = "6"
+    SEVEN = "7"
+
+
+class ValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPENING = "Opening"
+    CLOSED = "Closed"
+    CLOSING = "Closing"
+
+
+class FastValveState(str, Enum):
+    FAULT = "Fault"
+    OPEN = "Open"
+    OPEN_ARMED = "Open Armed"
+    CLOSED = "Closed"
+    CLOSED_ARMED = "Closed Armed"
+    NONE5 = "Unused"
+
+
+class LimitSwitchState(str, Enum):
+    OFF = "Off"
+    ON = "On"
+
+
+class AllValvesControl(StandardReadable):
```

Comment:
Name this after a specific piece of hardware or behaviour, so it's clearer what the intent is. 
From reading the device class, there's 8 valves,  of which 1,3 are slow and controlled by Epics, 5, 6 are fast and controlled by Epics and 2,4,7,8 may be manually (as in physically manipulated) set?

Do we need to distinguish between fast and slow moving valves in the read method? Are the slow valves changing not over the scale of the experiment but only between experiments? Presumably from the _ARMED enum, they're potentially hardware triggered?

Shouldn't the signals of type ValveControlRequest be SignalRW, the ControlRequest is the desired state?

Reply:
> Name this after a specific piece of hardware or behaviour, so it's clearer what the intent is

Apologies @barnettwilliam, this is contradictory to what I said in https://github.com/DiamondLightSource/dodal/pull/673#discussion_r1695332818. @DiamondJoseph and I are also arguing about the same thing over at https://github.com/DiamondLightSource/dodal/pull/713#discussion_r1698083552. I think I'm still not convinced. Is this something that core have generally agreed upon @DiamondJoseph or is it still up for debate?

Reply:
pressure jump cell is a i22 device we can worry about this specific naming if a moment comes when it is needed elsewhere.

Reply:
I think this [should be here](https://github.com/DiamondLightSource/dodal/pull/673#discussion_r1696889723), unless it's about AllValvesControl?

So far it's only me being a nuisance: looking at the (nicely newly refactored Ophyd-async) [StandardDetector impls](https://github.com/bluesky/ophyd-async/tree/main/src/ophyd_async/epics) (in `foo/_foo_io.py`) the pattern is consistent but probably because most examples have been copied from the AdAravis which initially had the pattern when it was thought we might need multiple Driver implementations for Mako/Mantis. 

I'm about ready to concede (and wouldn't have blocked merging on it) as examples from that repository are all being exported consistently to work as you have in your example on #713. I can just definitely see someone e.g. setting signal `A` and awaiting signal `A_at_final_position` (of type `A.FinalPositionReached` reaching `B.FinalPositionReached` and then that await times out and it's not clear debugging the scan what went wrong. Or being frustrated that type checking is highlighting an error "TypeError: Using type <long qualified path>.Foo, expected type <long qualified path>.Foo!"

Reply:
The problem you describe here @DiamondJoseph  is a function of the namespace pollution. 

I think we're fine for now for pressure cell valves and pumps as I haven't seen them in dodal.

You convinced me about the other though:
```
class BusyState(str, Enum):
    IDLE = "Idle"
    BUSY = "Busy"


class TimerState(str, Enum):
    TIMEOUT = "TIMEOUT"
    COUNTDOWN = "COUNTDOWN"


class StopState(str, Enum):
    CONTINUE = "CONTINUE"
    STOP = "STOP"
```
These *should* have the device name in them

Reply:
With regards to the device enums, does the underlying number matter or ever get used by ophyd/bluesky? should we make an effort to keep them aligned them with the  epics field numbers?

Reply:
which number?

Reply:
> I think this https://github.com/DiamondLightSource/dodal/pull/673#discussion_r1696889723, unless it's about AllValvesControl?

Yes, my bad, sorry

> These should have the device name in them

I've actually convinced @DiamondJoseph offline that they shouldn't have the device names prepended

>With regards to the device enums, does the underlying number matter or ever get used by ophyd/bluesky? should we make an effort to keep them aligned them with the epics field numbers?

The list of which strings are in the enum must match the list in EPICS but the order does not matter. This is enforced by a check when you run `dodal connect ` anyway.